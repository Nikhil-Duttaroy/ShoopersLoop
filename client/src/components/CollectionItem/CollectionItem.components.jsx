import React from 'react';
import './CollectionItem.styles.jsx'

import { connect } from "react-redux";

import { addItem } from '../../redux/cart/cart.actions';
// import useAlan from "../../hooks/useAlan.jsx";

import {
  CollectionItemContainer,
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer
} from './CollectionItem.styles';
import useAlan from '../../hooks/useAlan.jsx';

const CollectionItem = ({ item, addItem }) => {
  // useAlan();

  const { name, price, imageUrl, description ,tags,category} = item;
// useAlan(item)
  return (
    <CollectionItemContainer>
      <BackgroundImage className='image' imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
        {/* <PriceContainer>{tags}</PriceContainer> */}
      </CollectionFooterContainer>
      <AddButton onClick={() => addItem(item)} inverted>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  );
};

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(null, mapDispatchToProps)(CollectionItem);