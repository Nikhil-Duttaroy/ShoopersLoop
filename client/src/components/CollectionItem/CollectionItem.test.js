import React from "react";
import { shallow } from "enzyme";
import {CollectionItem} from './CollectionItem.components';

describe('CollectionItem Component', () => {
    let wrapper;
    let mockAddItem;
    const mockImageUrl = "www.test.com"
    const mockName="Black Shirt"
    const mockPrice=50

    beforeEach(()=>{
        mockAddItem=jest.fn();

        const mockProps={
            item:{
                imageUrl:mockImageUrl,
                name:mockName,
                price:mockPrice,
            },
            addItem:mockAddItem
        };
        wrapper = shallow(<CollectionItem {...mockProps} />);
    });


    it('should render CollectionItem', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it("should call addItem when Add Item Button is Clicked", () => {
        wrapper.find("AddButton").simulate("click");
        expect(mockAddItem).toHaveBeenCalled();
    });

    it("should render BackgroundImage with prop of ImageUrl", () => {
      expect(wrapper.find("BackgroundImage").prop("imageUrl")).toBe(mockImageUrl);
    });

    it("should render NameContainer with prop of name", () => {
      expect(wrapper.find("NameContainer").text()).toBe(mockName);
    });

    it("should render PriceContainer with prop of price", () => {
      const price = parseInt(wrapper.find("PriceContainer").text());
      expect(price).toBe(mockPrice);
    });






});
