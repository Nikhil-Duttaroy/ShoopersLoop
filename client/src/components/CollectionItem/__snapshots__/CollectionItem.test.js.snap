// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem Component should render CollectionItem 1`] = `
<styled.div>
  <BackgroundImage
    className="image"
    imageUrl="www.test.com"
  />
  <CollectionFooterContainer>
    <NameContainer>
      Black Shirt
    </NameContainer>
    <PriceContainer>
      50
    </PriceContainer>
  </CollectionFooterContainer>
  <AddButton
    inverted={true}
    onClick={[Function]}
  >
    Add to cart
  </AddButton>
  <ToastContainer
    autoClose={1500}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    theme="light"
    transition={[Function]}
  />
</styled.div>
`;
